### PR TITLE
chore: add husky pre-commit for formatting

### LINKED ISSUE #9: chore: add husky pre-commit for formatting
We could have pre-commit formatting with a certain configuration to avoid different formatting across developers.

---

**Tasks:**

- [x] Default eslint config
- [x] Default prettier config
- [x] husky pre-commit for formatting and linting
- [ ] lint-stage for staged files

### PR DESCRIPTION
**Pull request by CaseoJKL**
Adds husky pre-commit for linting and formatting and lint-staged for staged files.  closes #9

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,5 +23,11 @@
   "dependencies": {
     "alpinejs": "^3.13.0",
     "marked": "^8.0.1"
+  },
+  "lint-staged": {
+    "*.{js,jsx}": [
```
Comment (wiktoriavh) [inline package.json]: should support ts files, jsx can be removed.

File: tailwind.config.ts [hunk 2]
```diff
@@ -1,64 +1,64 @@
-import { Config } from 'tailwindcss';
```
Comment (wiktoriavh) [inline tailwind.config.ts:1]: i can see what you meant earlier in chat. I use single quote for the finished part, should update the eslint config then.

**Review by wiktoriavh — CHANGES_REQUESTED** [review]

File: .prettierrc.json [hunk 3]
```diff
@@ -1 +1,3 @@
-{}
+{
+  "singleQuote": true
```
Comment (wiktoriavh) [inline .prettierrc.json:2]: thanks!
You have to run it as well, to undo the previous changes.

**Review by wiktoriavh — CHANGES_REQUESTED** [review]

File: tailwind.config.ts (hunk 2 above)
Comment (CaseoJKL) [reply tailwind.config.ts:1]: I faced some errors I could not resolve from lint-staged when changing to ts. unexpected token }. and expected "from" on line 3 and some random errors irrelevant to the ts files. not sure why.

Comment (wiktoriavh) [reply tailwind.config.ts:1]: Can you tell me more about the error you get?

Comment (CaseoJKL) [reply tailwind.config.ts:1]: image

**Pull request merged**